### PR TITLE
Add crime incident report intent

### PR DESCRIPTION
**Pull request by jaumb**
Closes https://github.com/codeforboston/voiceapp311/issues/10

Very basic implementation so far.

### CODE REVIEW

File: mycity/mycity/test/integration_tests/test_crime_incident_intent.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        """
+        Patching out the functions in CrimeIncidentsIntent that use requests.get
+        """
+        super().setUp()
+        self.get_crime_incident_response = \
+            mock.patch(
+                ('mycity.intents.crime_activity_intent.'
+                 'get_crime_incident_response'),
+                return_value = test_constants.GET_CRIME_INCIDENTS_API_MOCK)
+        self.get_crime_incident_response.start()
```
Comment (jmartini) [inline mycity/mycity/test/integration_tests/test_crime_incident_intent.py]: Not required, but could be good to add some tests to make sure the `GET_CRIME_INCIDENTS_API_MOCK` is correctly parsed.

File: mycity/mycity/utilities/crime_incidents_api_utils.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+QUERY_LIMIT = 5
+CRIME_INCIDENTS_SQL_URL = \
+    "https://data.boston.gov/api/3/action/datastore_search_sql"
+
+logger = logging.getLogger(__name__)
+
+
+def get_crime_incident_response(address):
+    """
+    Executes the crime incident request and builds a text response
```
Comment (jmartini) [inline mycity/mycity/utilities/crime_incidents_api_utils.py]: Looks like this returns the raw json and doesn't build a text reponse

File: mycity/mycity/utilities/crime_incidents_api_utils.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    :param address: address to query
+    :return: a response object
```
Comment (jmartini) [inline mycity/mycity/utilities/crime_incidents_api_utils.py]: I don't thin this return doc string is correct.

File: mycity/mycity/utilities/gis_utils.py [hunk 4]
```diff
@@ -110,3 +113,14 @@ def _get_dest_addresses_from_features(feature_address_index, features):
             dest_address += " Boston, MA"
             dest_addresses.append(dest_address)
     return dest_addresses
+
+
+def geocode_address(m_address):
```
Comment (jmartini) [inline mycity/mycity/utilities/gis_utils.py]: What does the `m_` prefix stand for here?

Comment (jaumb) [reply mycity/mycity/utilities/gis_utils.py]: _My address? I have no idea, really. I borrowed this code from another pull request but based on tonight's discussion this function will be replaced by one that calls the rest API directly.

File: mycity/mycity/intents/crime_activity_intent.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+            _build_text_from_response(response)
+    else:
+        logger.debug("Error: Called crime_incidents_intent with no address")
+
+    # Setting reprompt_text to None signifies that we do not want to reprompt
+    # the user. If the user does not respond or says something that is not
+    # understood, the session will end.
+    mycity_response.reprompt_text = None
+    mycity_response.session_attributes = mycity_request.session_attributes
+    mycity_response.card_title = mycity_request.intent_name
```
[the remark this answers is not in the document]
Comment (alidaniell) [reply mycity/mycity/intents/crime_activity_intent.py]: In keeping with the way some of the other intents (trash_day, feedback, get_alerts, latest_311) do this, suggest change line from:

 mycity_response.card_title = mycity_request.intent_name

to

 mycity_response.card_title = CARD_TITLE_CRIME

and add line assigning title, as below, at top of crime_activity_intent.py file following import and logging statements and before first function is defined. 

CARD_TITLE_CRIME = "Crime Report"

File: mycity/mycity/test/integration_tests/test_crime_incident_intent.py (hunk 1 above)
Comment (jaumb) [reply mycity/mycity/test/integration_tests/test_crime_incident_intent.py]: Added a check that the response contains the the address

**Comment by jaumb** [comment]
@jmartini I made those adjustments, will you please take another look when you have a chance?

File: mycity/mycity/utilities/gis_utils.py (hunk 4 above)
Comment (thiteixeira) [reply mycity/mycity/utilities/gis_utils.py]: sorry to jump in, but `m_` means member variable. It's commonly used in C++ (and Java I guess)

**Pull request merged**